[DOCS-1776] Synthetics ICMP Tests (#9463)
DataDog/documentation: content/en/synthetics/api_tests/icmp_tests.md

---
title: ICMP Tests
kind: documentation
description: Monitor availability of your hosts and diagnose network issues
aliases:
  - /synthetics/icmp_test
  - /synthetics/icmp_check
further_reading:
- link: "https://www.datadoghq.com/blog/introducing-synthetic-monitoring/"
  tag: "Blog"
  text: "Introducing Datadog Synthetic Monitoring"
- link: "/getting_started/synthetics/api_test"
  tag: "Documentation"
  text: "Get started with API tests"
- link: "/synthetics/private_locations"
  tag: "Documentation"
  text: "Run ICMP Pings on internal endpoints"
---

## Overview

> ICMP tests allow you to easily **monitor the availability of your hosts and diagnose network communication issues to verify responsiveness from your application's endpoints**. By asserting on the values received from one or more **ICMP ping(s)** to your endpoint, Datadog can help **detect availability/connectivity issues, above-quota latency or hops, and changes in security firewall configuration**. ICMP tests can run from [managed][1] and [private locations][2] depending on whether you are willing to trigger ICMP pings to your endpoints from outside or inside your network.

Reviewer: ```suggestion
ICMP tests allow you to easily **monitor the availability of your hosts** and **diagnose network communication issues**. By asserting on the values received from one or more **ICMP ping(s)** to your endpoint, Datadog can help **detect connectivity issues, above-quota latency or hops, and unexpected changes in security firewall configuration**.  
 
ICMP tests can run from [**managed**][1] and [**private locations**][2] depending on whether you are willing to trigger ICMP pings to your endpoints from outside or inside your network.
```

Can you also add the bit about the type of runs? (ie. scheduled, on demand, etc.).

## Configuration

After choosing the type of test you are willing to create ([`HTTP`][3], [`SSL`][4], [`TCP`][5], [`DNS`][6], or [`ICMP` test][7]), you can define your test's request.

### Define Request

> ![][8]

Reviewer: Fix screenshot

> 1. Specify the **domain name** or **IP address** to run your test on.

Reviewer: ```suggestion
1. Specify the **Domain Name** or **IP address** to run your test on.
```

> 2. Select the **number of pings** to trigger per test session. By default, the number of pings is set to 4 - you can choose to increase or decrease.

Reviewer: ```suggestion
2. Select the **Number of Pings** to trigger per test session. By default, the number of pings is set to 4 - you can choose to increase or decrease.
```

3. **Name** your ICMP test.

> 4. Add env **Tags** as well as any other tag to your ICMP test. You can then use these tags to quickly filter through your Synthetic tests on the [Synthetic Monitoring homepage][9].

Reviewer: ```suggestion
4. Add `env` **Tags** as well as any other tag to your ICMP test. You can then use these tags to quickly filter through your Synthetic tests on the [Synthetic Monitoring homepage][9].
```

5. Select the **Locations** to run your ICMP test from: ICMP tests can run from [managed][1] and [private locations][2] depending on whether you are willing to trigger the ICMP pings from outside or inside your network.

Click on **Test URL** to try out the request configuration. You will see a response preview show up on the right side of your screen.

### Specify test frequency

ICMP tests can run:

> * **On a schedule** to ensure your most important services are always accessible to your users. Select the frequency you want Datadog to run your ICMP test.

Reviewer: Add the screenshot that is used in other tests?

* [**Within your CI/CD pipelines**][10].
* **On-demand** to run your tests whenever makes the most sense for your teams.

### Define Assertions

> Assertions define what an expected test result is. When hitting `Test URL` basic assertions on `latency`, `packet loss` and ICMP ping result is added. You must define at least one assertion for your test to monitor.

Reviewer: ```suggestion
Assertions define what an expected test result is. When hitting `Test URL` basic assertions on `latency`, `packet loss` and `ICMP ping result` is added. You must define at least one assertion for your test to monitor.
```

| Type          | Operator                                                                               | Value Type       |
|---------------|----------------------------------------------------------------------------------------|------------------|

> | latency       | `is less than`, `is less than or equal`, `is`, `is more than`, `is more than or equal` | _integer (ms)_   |

Reviewer: This table is lacking the aggregation method bit for the latency assertions

| packet loss   | `is less than`, `is less than or equal`, `is`, `is more than`, `is more than or equal` | _percentage (%)_ |
| ICMP ping     | `Successful` or `Failed`                                                               | _string_         |

> You can create up to 10 assertions per API test by clicking on **New Assertion** or by clicking directly on the response preview:

Reviewer: Add screenshot

    |----------------------------|---------------------------------------------------------------------|
    | `{{#is_alert}}`            | Show when the test alerts.                                          |
    | `{{^is_alert}}`            | Show unless the test alerts.                                        |
    | `{{#is_recovery}}`         | Show when the test recovers from alert.                             |
    | `{{^is_recovery}}`         | Show unless the test recovers from alert.                           |

3. Specify how often you want your test to **re-send the notification message** in case of test failure. To prevent renotification on failing tests, leave the option as `Never renotify if the monitor has not been resolved`.

Email notifications include the message defined in this section as well as a summary of failed assertions.
Notifications example:

{{< img src="synthetics/api_tests/notifications-example.png" alt="API Test Notifications"  style="width:90%;" >}}

Click on **Save** to save your test and have Datadog start executing it.

## Variables

### Create local variables

> You can create local variables by defining their values from one of the below available builtins:

Reviewer 2: ```suggestion
You can create local variables by defining their values from one of the below available built-ins:
```

| `{{ numeric(n) }}`         | Generates a numeric string with `n` digits.                                                                 |
| `{{ alphabetic(n) }}`      | Generates an alphabetic string with `n` letters.                                                            |
| `{{ alphanumeric(n) }}`    | Generates an alphanumeric string with `n` characters.                                                       |
| `{{ date(n, format) }}`    | Generates a date in one of our accepted formats with a value of the date the test is initiated + `n` days.        |
| `{{ timestamp(n, unit) }}` | Generates a timestamp in one of our accepted units with a value of the timestamp the test is initiated at +/- `n` chosen unit. |

### Use variables

You can use the [global variables defined in the `Settings`][13] and the [locally defined variables](#create-local-variables) in the URL and assertions of your ICMP tests.
To display your list of variables, type `{{` in your desired field:

{{< img src="synthetics/api_tests/use_variable.mp4" alt="Using Variables in API tests" video="true" width="90%" >}}

## Test failure

> A test is considered `FAILED` if it does not satisfy one or several assertions or if the request prematurely failed. In some cases, the test can indeed fail without being able to test the assertions against the endpoint, these reasons include:

Reviewer 2: ```suggestion
A test is considered `FAILED` if it does not satisfy one or several assertions or if the request prematurely failed. In some cases, the test can fail without being able to test the assertions against the endpoint, these reasons include:
```

| DNS               | DNS entry not found for the test URL. Possible causes include misconfigured test URL, wrong configuration of your DNS entries, etc.                                                                                                                                                                                                                                                                                                                  |
| `INVALID_REQUEST` | The configuration of the test is invalid (for example, a typo in the URL).                                                                                                                                                                                                                                                                                                                                                                                     |

> | `SSL`             | The SSL connection couldn't be performed. [See the dedicated error page for more information][8].                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: Let's completely get rid of the SSL line - I recently removed it on DNS & TCP tests :)
